pkg/pg/collectors.go

```diff
@@ -30,7 +30,33 @@ AS f
 func PGStatStatements(pgPool *pgxpool.Pool) func(ctx context.Context, state *agent.MetricsState) error {
 	return func(ctx context.Context, state *agent.MetricsState) error {
 		var jsonResult string
-		err := pgPool.QueryRow(ctx, PgStatStatementsQuery).Scan(&jsonResult)
+		var err error
+
+		// Get config
+		pgConfig, _ := ConfigFromViper(nil)
```

Comment:
This gets called every single iteration, you can move it outside of the `return func()` and it will get run once and be included inside the functions context

Reply:
I pushed the fix